stories/__tests__/SQFormIconButton.stories.test.js

```diff
@@ -0,0 +1,153 @@
+import {composeStories} from '@storybook/testing-react';
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import * as stories from '../SQFormIconButton.stories';
+
+const {
+  Default: SQFormIconButton,
+  WithTestField: SQFormIconButtonWithField
+} = composeStories(stories);
+
+describe('SQFormIconButton Tests', () => {
+  describe('Button Only', () => {
+    it('should render a submit button', () => {
+      render(<SQFormIconButton exampleIcons={CheckCircle} />);
+
+      const iconButton = screen.getByRole('button', {name: /form submission/i});
+
+      expect(iconButton).toBeInTheDocument();
+    });
+
+    it('should render a reset button given the type reset', () => {
+      render(<SQFormIconButton exampleIcons={CheckCircle} type="reset" />);
+
+      const iconButton = screen.getByRole('button', {
+        name: /form submission/i
+      });
+
+      expect(iconButton).toBeInTheDocument();
+      expect(iconButton).toHaveAttribute('type', 'reset');
+    });
+
+    it('should render a button given the type button', () => {
+      render(<SQFormIconButton exampleIcons={CheckCircle} type="button" />);
+
+      const iconButton = screen.getByRole('button', {
+        name: /form submission/i
+      });
+
+      expect(iconButton).toBeInTheDocument();
+      expect(iconButton).toHaveAttribute('type', 'button');
+    });
+
+    it('should call a function when clicked', () => {
+      const onClickSpy = jest.fn();
+      render(
+        <SQFormIconButton exampleIcons={CheckCircle} onClick={onClickSpy} />
+      );
+
+      const iconButton = screen.getByRole('button', {
+        name: /form submission/i
+      });
+
+      userEvent.click(iconButton);
+
+      expect(onClickSpy).toHaveBeenCalled();
+    });
+
+    it('should show as disabled when isDisabled is true', () => {
+      render(<SQFormIconButton exampleIcons={CheckCircle} isDisabled={true} />);
+
+      const iconButton = screen.getByRole('button', {
+        name: /form submission/i
+      });
+
+      expect(iconButton).toBeDisabled();
+    });
+
+    it('should not fire onClick when disabled', () => {
+      const onClickSpy = jest.fn();
+      render(
+        <SQFormIconButton
+          exampleIcons={CheckCircle}
+          isDisabled={true}
+          onClick={onClickSpy}
+        />
+      );
+
+      const iconButton = screen.getByRole('button', {
+        name: /form submission/i
+      });
+
+      userEvent.click(iconButton);
+
+      expect(onClickSpy).not.toHaveBeenCalled();
+    });
+
+    it('should disable when shouldRequireFieldUpdates is true', () => {
+      render(
+        <SQFormIconButton
+          exampleIcons={CheckCircle}
+          shouldRequireFieldUpdates={true}
+        />
+      );
+
+      const iconButton = screen.getByRole('button', {
+        name: /form submission/i
+      });
+
+      expect(iconButton).toBeDisabled();
+    });
+  });
+
+  describe('Button with Test Field', () => {
```

Comment:
typo: `text`

Reply:
I meant it to be Test. The name of the story is `WithTestField`